src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -561,11 +569,9 @@ DeferErrorIfUnsupportedSubqueryPushdown(Query *originalQuery,
 	else if (!RestrictionEquivalenceForPartitionKeys(plannerRestrictionContext))
 	{
 		return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
-							 "cannot pushdown the subquery since all relations are not "
-							 "joined using distribution keys",
-							 "Each relation should be joined with at least "
-							 "one another relation using distribution keys and "
-							 "equality operator.", NULL);
+							 "complex joins are only supported when all distributed tables are "
```

Comment:
As I wrote somewhere else, we should probably want to use `distribution column` instead of `partition columns` to prevent any confusion Postgres partitioning vs Citus.

And, we'd also want to mention that the joins should be `equi joins / equality operator` etc.

Reply:
done